node/modules/actorevent.go

```diff
@@ -111,6 +146,7 @@ func EthEventAPI(cfg config.FevmConfig) func(helpers.MetricsCtx, repo.LockedRepo
 					return address.Undef, false
 				}
 				// we have an f4 address, make sure it's assigned by the EAM
+				// What happens when we introduce events for built-in Actor events here ?
```

Comment:
It's better to use a shared FilterManager/Indexer between the ETH events API and this new API.

However, the existing address resolver in the ETH API has this check which I believe will fail for built-in actor addresses. Can we remove this check if we're going to share this resolver b/w the two APIs or should we refactor the code so that both APIs can pass in their own address resolvers ?

Reply:
I'd propose:
- preferring f4 addresses where possible (this is a fast check anyway)
- if no f4 address found, try to find an f1 / f3 address in the init actor (this can be left as a TODO for now, since IIUC accounts will never be emitting events)
- return ID otherwise

I think this should get what we want without changing any existing Eth behaviour, but this is more a @Stebalien department.

Reply:
Honestly, I think this code is just wrong.

1. We should record the _ID_ in the index (resolve nothing). The ActorID plus the tipset is the "canonical" identifier for the actor.
2. When querying, resolve addresses to IDs, then query. As long as we limit ourselves to querying for non-reverted events, there shouldn't be any issue of ID reuse.
3. When subscribing, we'll have to resolve the filter's addresses to IDs at every tipset, but that shouldn't be a huge deal.

Of course... that'll likely require a migration. A short-term solution is to just use the `actor.Address` if specified (drop the "is delegated" check), otherwise, use the ID. That should "work"-ish, but won't be great because it, e.g., won't work for f2 addresses.

As @arajasek points out, we can punt on f1/f3 for now. However, IMO, we should punt on that forever and just "fix" this the correct way by indexing by ID.

Reply:
Following @arajasek 's suggestion here for now and returning ID Addresses if we can't find a robust/f4 address for the given Actor ID.

Reply:
@Stebalien Why wont it work for f2 addresses ?

Reply:
Because we don't have a table mapping IDs to f2 addresses and maintaining one would be annoying (given reverts, etc.). So, it _can_ be done, but it's inefficient (you'd need to scan the entire address resolution table in the init actor for each lookup).